core/src/packet_sender_info.rs

```diff
@@ -0,0 +1,56 @@
+/// struct PacketSenderInfo contains sender information for pakcets in baning_stage's
+/// unprocessed_buffer. It provides metrics for stake weight block packing prioritization.
+///
+use std::{collections::HashMap, net::IpAddr};
+
+#[derive(Debug, Default)]
+pub struct SenderDetailInfo {
+    pub stake: u64,
+    pub packet_count: u64,
+}
+
+#[derive(Debug, Default)]
+pub struct PacketSenderInfo {
+    // Vec of IP of packets in banking_stage buffer, one-to-one mapped, even after shuffling.
+    pub packet_senders_ip: Vec<IpAddr>,
+    // HashMap keyed by IP, with sender's detail info as Value
+    pub senders_detail: HashMap<IpAddr, SenderDetailInfo>,
+    // implied info are:
+    // 1. packet_senders_ip.len() is the number of buffered packets subjected to this round of
+    //    processing;
+    // 2. sender_detail.values.map(|info| info.stake).sum() is the total stakes presented in buffer
```

Comment:
nit: not clear what this `buffer` is referring to. 

Is this the total *unique* stake in the banking stage unprocessed queue? 

Or is stake from the same validator counted multiple times?